src/Symfony/Component/DependencyInjection/ContainerInterface.php

```diff
@@ -37,7 +37,9 @@ public function set(string $id, ?object $service): void;
      *
      * @param B $invalidBehavior
      *
-     * @psalm-return (B is self::EXCEPTION_ON_INVALID_REFERENCE|self::RUNTIME_EXCEPTION_ON_INVALID_REFERENCE ? object : object|null)
+     * @psalm-return ($id is class-string<A> ? (B is self::RUNTIME_EXCEPTION_ON_INVALID_REFERENCE|self::EXCEPTION_ON_INVALID_REFERENCE ? A : A|null) : (B is self::RUNTIME_EXCEPTION_ON_INVALID_REFERENCE|self::EXCEPTION_ON_INVALID_REFERENCE ? object : object|null))
+     *
+     * @return ($id is class-string<A> ? (B is 0|1 ? A : A|null) : (B is 0|1 ? object : object|null))
```

Comment:
not sure we want conditional types on @return
using @psalm-return should be enough, isn't it?